### ISSUE TITLE
Allow deselect/hide a subfield of a parent/compound field

### ISSUE THREAD

**Issue by kamil386** [IQSS/dataverse#6588]
Allow Dataverse user to hide/deselect the subfield, so that dataset depositors don't see that field in the "parent" or compound field.

The reason is that a depositor could be overwhelmed by the number of fields that she may not need to be concerned about, especially for the big schemas like DarwinCore.

Related to #6243

**Comment by pdurbin** [comment]
@kamil386 do you have a screenshot handy of what the DarwinCore metadata block looks like when it's being edited? And if so, would you be able to give an example of a child/subfield of a compound field that you might want to hide?

**Comment by kamil386** [comment]
I would like to hide any of these 34 subfields of Taxon group... I don;t know how many fields the scientists will choose, but probably it will depend on the biodiversity object.

![image](https://user-images.githubusercontent.com/56053188/73448454-2af7e800-4361-11ea-9dfd-20031c795892.png)

**Comment by pdurbin** [comment]
@kamil386 yes, I agree that having so many options for IDs is overwhelming. Thanks for the screenshot!

**Comment by djbrooke** [comment]
@kamil386 apologies if I'm missing something, but can you provide some more information about why you'd include these fields in the metadata block if you don't plan to collect this information? I guess I don't understand what's being asked here.  

cc: @jggautier

**Comment by kamil386** [comment]
@djbrooke I think it's not a good idea if everyone will add to the Dataverse their own customized DwC TSV metadata block with individual chosen set of fields and subfields. It'll lead to mess. What is the chance that we'll select the optimal set of fields from DwC schema that will fit for every biodiversity portals all over the world? What if someone will need some other fields that we don't care about?

I think that better approach is to implement full DwC specification (that has been already done) and allow users to choose (hide/deselect some of the fields) during dataverse creation the appropriate set of fields on their own.

Of course, we we'll finally choose our own set of fields from DwC, but I think community and other biodiversity portal needs more flexible solution.

**Comment by cmbz** [comment]
2024/09/30: Keeping as it is a useful feature, though may only be relevant for Darwin Core.

**Comment by qqmyers** [comment]
FWIW: It might also be related to the issues around making some fields recommended, e.g. recommended fields are the only ones that show by default and truly optional ones hare hidden by default (unless perhaps the user clicks an 'advanced' button.)
